src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    public static String h_rule = "────────────────────────────────────────────────────────────────────";
+
+    public static int dukeRunner(String log) throws DukeNotFoundException, DukeDescriptionException, DukeTimingException, FileNotFoundException {
```

Comment:
Since there are some many different types of exceptions the method may throw, maybe it is better to create a parent class DukeException of all of them and the method header may be shorter. e.g. `public static int dukeRunner(String log) throws DukeException {`